eZ/Publish/API/Repository/ContentTypeService.php

```diff
@@ -155,11 +155,12 @@ public function loadContentTypeByRemoteId($remoteId, array $prioritizedLanguages
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If the content type draft owned by the current user can not be found
      *
-     * @param mixed $contentTypeId
+     * @param int $contentTypeId
```

Comment:
I think it should be sadly reverted to `mixed` to be consistent across docs.

Reply:
> I think it should be sadly reverted to `mixed` to be consistent across docs.

Absolutely not. Please forget about `mixed`, it's a disease of PHP. There's a limited number of cases where it makes sense (due to lack of variable templating). Usually it's a mistake.
If anything, other docs should be aligned.

But please elaborate on consistency, where and why. Maybe I missed something.

Reply:
>other docs should be aligned.

thats why. I think it is simply wrong in same class in one place have typed `int` and in other `mixed` for same thing. As I agree `mixed` is bad I think we should either fix them all or leave them as they are.

Reply:
> > other docs should be aligned.
> 
> thats why. I think it is simply wrong in same class in one place have typed `int` and in other `mixed` for same thing. As I agree `mixed` is bad I think we should either fix them all or leave them as they are.

Well... other docs are also part of the diff, so it doesn't hurt too much to change in all implementations of this method. Probably should be a separate PR, but seems to me like an overkill.

Reply:
It should be typehinted as `int|string` instead of `mixed`. eZ Repository API from the beginning in theory allows identifiers to be strings.

Reply:
> It should be typehinted as `int|string` instead of `mixed`. eZ Repository API from the beginning in theory allows identifiers to be strings due to possibility of different storage engines where some might use strings as identifiers.

Those are mistakes of the past. Mixed typing advocates worst possible software engineering practices, being also very error prone. The direction PHP is heading to (strict typing) also confirms that.

For the use case you've just described, the SOLID solution would be to delegate proper transformation or mapping of identifier to a specific storage implementation. In this case `id` on the application/API level would be a ValueObject ready for different strategies.

Reply:
I don't agree. Those were not mistakes of the past, but rather the best it could've been done in that time period and constraints.

In any case I much more appreciate consistency than having *different* typehints for the *same* value in *different* places (even within the same file). It brings in much more mess than just having everything consistent and leaves a sour taste in mouth of authors being careless.